tests/subsys/llext/testcase.yaml

```diff
@@ -3,12 +3,19 @@ common:
 tests:
   llext.simple.arm:
     filter: not CONFIG_CPU_HAS_MMU and not CONFIG_SOC_SERIES_S32ZE_R52
+      and not CONFIG_SOC_FAMILY_STM32
     arch_allow: arm
     extra_configs:
       - CONFIG_ARM_MPU=n
     # Broken platforms
     platform_exclude:
       - nuvoton_pfm_m487 # See #63167
+  llext.simple.arm.stm32:
+    filter: not CONFIG_CPU_HAS_MMU and CONFIG_SOC_FAMILY_STM32
+    arch_allow: arm
+    extra_configs:
+      - CONFIG_ARM_MPU=n
+      - CONFIG_CACHE_MANAGEMENT=y
```

Comment:
That is fine for the tests. But should it be extended in a generic fashion when llext is used on these targets?
Is that specific to these targets because they hold a cache ? Are they the only target with cache having this issue?

It is nice to fix the test, but if it reveals an issue, I'd like to be sure the actual issue is fixed once for good.

Reply:
Yes, the issue is indeed generic to any architecture with dual I/D caches and no coherency guarantees.
 
I'm not sure how to add this as a requirement to use the llext subsystem, though. Can we add a `select CONFIG_CACHE_MANAGEMENT` in our subsys Kconfig even on arches that don't define the symbol?

Reply:
You can do for instance `select CONFIG_CACHE_MANAGEMENT if CPU_HAS_DCACHE`

Reply:
That should be great, thanks!  (I left a CONFIG_ in there, it should be 
```
select CACHE_MANAGEMENT if CPU_HAS_DCACHE
```
).

Reply:
@pillo79 Do you plan to push a PR to fix this ?

Reply:
I just did as part of #66476. 
I ended up with an `if DCACHE` so `CACHE_MANAGEMENT` doesn't get enabled if the user disables the whole cache interface.